netty/src/main/java/io/grpc/transport/netty/GrpcSslContexts.java

```diff
@@ -93,7 +111,44 @@ public static SslContextBuilder forServer(
    * an application requires particular settings it should override the options set here.
    */
   public static SslContextBuilder configure(SslContextBuilder builder) {
-    return builder.ciphers(Http2SecurityUtil.CIPHERS, SupportedCipherSuiteFilter.INSTANCE)
-        .applicationProtocolConfig(DEFAULT_APN);
+    return configure(builder, SslContext.defaultServerProvider());
+  }
+
+  /**
+   * Set ciphers and APN appropriate for gRPC. Precisely what is set is permitted to change, so if
+   * an application requires particular settings it should override the options set here.
+   */
+  public static SslContextBuilder configure(SslContextBuilder builder, SslProvider provider) {
+    // The default client/server providers in Netty are the same, so it doesn't matter which
+    // method we call.
+    return builder.sslProvider(provider)
```

Comment:
> I'm not sure I follow your concern. Are you uneasy because we're calling a method SslContext.defaultServerProvider() for both client and server?

I'm not concerned about the method name. I'm concerned that the default client and server providers aren't required to match in Netty's SslContext API. Netty can change the default for client and server independently.

> I've copied the generic defaultProvider code into this class to get rid of that amgibuity.

I guess that works, because we are setting `sslProvider()` in `configure()`.

However, we don't actually work with OpenSSL today, so maybe `defaultSslProvider()` should always return JDK, and `selectApplicationProtocolConfig()` should throw an exception for OpenSSL?

> As far as allowing the application to specify whether they want to use OpenSSL or not, I think we should provide the option regardless of whether or not OpenSSL is installed.

Why is that? Why let them use it even if it is known-broken?

> They may want to use JDK even though OpenSSL is supported, for example

That's something different. Yes, letting them choose between the various working implementations is a fine goal.

> (e.g. maybe they want to enforce ALPN

But that's something different. I'm not sure we need such an API, at least right now.